MatrixSDK/Aggregations/LocationSharing/MXBeaconAggregations.swift

```diff
@@ -0,0 +1,170 @@
+            return
+        }
+        
+        guard let beaconInfo = MXBeaconInfo(mxEvent: event) else {
+            return
+        }
+        
+        self.addOrUpdateBeaconInfo(beaconInfo, inRoomWithId: roomId)
+    }
+    
+    // MARK: Data update listener
+    
+    public func listenToBeaconInfoSummaryUpdateInRoom(withId roomId: String, handler: @escaping (MXBeaconInfoSummaryProtocol) -> Void) -> Any? {
+        let listener = MXBeaconInfoSummaryListener(roomId: roomId, notificationHandler: handler)
+        
+        listeners.append(listener)
+
+        return listener
+    }
+
+    public func removeListener(_ listener: Any) {
+        guard let beaconInfoSummaryListener = listener as? MXBeaconInfoSummaryListener else {
+            return
+        }
+        
+        listeners.removeAll(where: { $0 === beaconInfoSummaryListener })
+    }
+    
+    // MARK: - Private
+    
+    private func addOrUpdateBeaconInfo(_ beaconInfo: MXBeaconInfo, inRoomWithId roomId: String) {
+        
+        guard let eventId = beaconInfo.originalEvent?.eventId else {
+            return
+        }
+        
+        var beaconInfoSummary: MXBeaconInfoSummary?
+        
+        if let existingBeaconInfoSummary = self.getBeaconInfoSummary(withIdentifier: eventId, inRoomWithId: roomId) {
+
+            // If beacon info is older than existing one, do not take it into account
+            if beaconInfo.timestamp > existingBeaconInfoSummary.beaconInfo.timestamp {
+                existingBeaconInfoSummary.updateWithBeaconInfo(beaconInfo)
+                beaconInfoSummary = existingBeaconInfoSummary
+            }
+        } else {
+            beaconInfoSummary = MXBeaconInfoSummary(beaconInfo: beaconInfo)
+        }
+        
+        if let beaconInfoSummary = beaconInfoSummary {
+            self.beaconInfoSummaryStore.addOrUpdateBeaconInfoSummary(beaconInfoSummary, inRoomWithId: roomId)
+            
+            self.notifyBeaconInfoSummaryListeners(ofRoomWithId: roomId, beaconInfoSummary: beaconInfoSummary)
+        }
+    }
+    
+    private func canAddBeacon(_ beacon: MXBeacon, to beaconInfoSummary: MXBeaconInfoSummary) -> Bool {
+    
+        guard beaconInfoSummary.hasStopped == false, beaconInfoSummary.hasExpired == false,
+        beacon.timestamp < beaconInfoSummary.expiringDate else {
+            return false
+        }
+        
+        if let lastBeacon = beaconInfoSummary.lastBeacon, beacon.timestamp < lastBeacon.timestamp {
+            return false
+        }
+        
+        return true
+    }
+    
+    private func notifyBeaconInfoSummaryListeners(ofRoomWithId roomId: String, beaconInfoSummary: MXBeaconInfoSummary) {
+        
+        for listener in listeners where listener.roomId == roomId {
+            listener.notificationHandler(beaconInfoSummary)
+        }
+    }
+    
+    private func getBeaconInfoSummary(withIdentifier identifier: String, inRoomWithId roomId: String) -> MXBeaconInfoSummary? {
```

Comment:
The difference between `getBeaconInfoSummary` and `beaconInfoSummary` wasn't obvious at first glance, although I'm not sure I have any suggestions on better naming 😕